hugo/assets/scripts/statemachine.tsx

```diff
@@ -0,0 +1,356 @@
+        <div className="text-white border-2 border-solid border-accentRed rounded-md p-4 text-left text-sm cursor-default">
+        {this.state.diagnostics.map((diagnostic, index) => 
+          <details key={index}>
+            <summary>{`Line ${diagnostic.range.start.line}-${diagnostic.range.end.line}: ${diagnostic.message}`}</summary>
+            <p>Source: {diagnostic.source} | Code: {diagnostic.code}</p>
+          </details>
+        )}
+        </div>
+      </div>
+    );
+  }
+}
+
+class App extends React.Component<{}> {
+  monacoEditor: React.RefObject<MonacoEditorReactComp>;
+  preview: React.RefObject<Preview>;
+  constructor(props) {
+    super(props);
+
+    // bind 'this' ref for callbacks to maintain parent context
+    this.onMonacoLoad = this.onMonacoLoad.bind(this);
+    this.onDocumentChange = this.onDocumentChange.bind(this);
+    this.monacoEditor = React.createRef();
+    this.preview = React.createRef();
+  }
+
+  /**
+   * Callback that is invoked when Monaco is finished loading up.
+   * Can be used to safely register notification listeners, retrieve data, and the like
+   *
+   * @throws Error on inability to ref the Monaco component or to get the language client
+   */
+  onMonacoLoad() {
+    // verify we can get a ref to the editor
+    if (!this.monacoEditor.current) {
+      throw new Error("Unable to get a reference to the Monaco Editor");
+    }
+
+    // verify we can get a ref to the language client
+    const lc = this.monacoEditor.current
+      ?.getEditorWrapper()
+      ?.getLanguageClient();
+    if (!lc) {
+      throw new Error("Could not get handle to Language Client on mount");
+    }
+
+    // register to receive DocumentChange notifications
+    lc.onNotification("browser/DocumentChange", this.onDocumentChange);
+  }
+
+  /**
+   * Callback invoked when the document processed by the LS changes
+   * Invoked on startup as well
+   *
+   * @param resp Response data
+   */
+  onDocumentChange(resp: DocumentChangeResponse) {
+    // decode the received Ast
+    const statemachineAst = new LangiumAST().deserializeAST(resp.content) as StateMachineAstNode;
+    this.preview.current?.startPreview(statemachineAst, resp.diagnostics);
+  }
+
+  render() {
+    const style = {
+      paddingTop: "5px",
+      height: "100%",
+      width: "100%",
+    };
+
+    return (
+      <div className="w-full h-full border border-emeraldLangium justify-center self-center flex">
+        <div className="float-left w-1/2 h-full border-r border-emeraldLangium">
+          <div className="wrapper relative bg-white dark:bg-gray-900">
+            <MonacoEditorReactComp
+              ref={this.monacoEditor}
+              onLoad={this.onMonacoLoad}
+              webworkerUri="../showcase/libs/worker/statemachineServerWorker.js"
+              workerName="LS"
+              workerType="classic"
+              languageId="statemachine"
+              text={`// Create your own statemachine here!
+statemachine TrafficLight
```

Comment:
Move that giant string to a own file or at least a variable. I disturbs a bit while reading the code.